tests/model_serving/model_server/authentication/utils.py

```diff
@@ -0,0 +1,111 @@
+import base64
+import re
+from contextlib import contextmanager
+from typing import Dict, List, Optional
+
+from kubernetes.dynamic import DynamicClient
+from ocp_resources.inference_service import InferenceService
+from ocp_resources.role import Role
+from simple_logger.logger import get_logger
+
+from utilities.inference_utils import Inference
+
+LOGGER = get_logger(name=__name__)
+
+
+def base64_encode_str(text: str) -> str:
+    return base64.b64encode(text.encode()).decode()
+
+
+def verify_inference_response(
+    inference_service: InferenceService,
+    runtime: str,
+    inference_type: str,
+    protocol: str,
+    model_name: str,
+    inference_text: str,
+    expected_response_text: Optional[str] = "",
+    insecure: bool = True,
+    token: Optional[str] = None,
+    authorized_user: Optional[bool] = None,
+) -> None:
+    inference = Inference(
+        inference_service=inference_service,
+        runtime=runtime,
+        inference_type=inference_type,
+        protocol=protocol,
+    )
+
+    res = inference.run_inference(
+        model_name=model_name,
+        text=inference_text,
+        token=token,
+        insecure=insecure,
+    )
+
+    if authorized_user is False:
+        auth_header = "x-ext-auth-reason"
+
+        if auth_reason := re.search(rf"{auth_header}: (.*)", res["output"], re.MULTILINE):
+            reason = auth_reason.group(1).lower()
+
+            if token:
+                assert re.search(r"not (?:authenticated|authorized)", reason)
+
+            else:
+                assert "credential not found" in reason
+
+        else:
+            LOGGER.error(f"Auth header {auth_header} not found in response. Response: {res['output']}")
+            raise
+
+    elif inference.inference_response_text_key_name:
+        assert res["output"][inference.inference_response_text_key_name] == expected_response_text
+
+    elif output := re.findall(r"generated_text\": \"(.*)\"", res["output"], re.MULTILINE):
+        assert "".join(output) == expected_response_text
+
+    else:
+        LOGGER.error(f"Inference response text not found in response. Response: {res}")
+        raise
+
+
+def get_s3_secret_dict(
```

Comment:
Already defined in https://github.com/opendatahub-io/opendatahub-tests/pull/22/files/38b1f728c8d82955805212d36cf8c47f03d17a10#diff-ba4079b730a33a115ca68082ee9cdafe3a1d4cbff3aba9d09d508deacafbb183R51

Reply:
Consider moving to `utilities/infra.py`

Reply:
we should separate util functions from resource-related functions. under "infra" we have resource stuff.

Reply:
moved to utlities/general.py